File: from-2801-to-2900/2874-maximum-value-of-an-ordered-triplet-ii/solution.py
class Solution:
    def maximumTripletValue(self, nums: List[int]) -> int:
        maxi = float('-inf')
        diff = 0
        result = 0
        
        for i in range(len(nums)):
            maxi = max(maxi, nums[i])
            if i >= 2:
                result = max(result, diff * nums[i])
            if i >= 1:
                diff = max(diff, maxi - nums[i])
                
        return result

File: from-2801-to-2900/2874-maximum-value-of-an-ordered-triplet-ii/solution.go
func maximumTripletValue(nums []int) int64 {
    var result int64 = 0 
    maxi, diff := -1, 0
    n := len(nums)
    for i := 0; i < n; i ++ {
        maxi = max(maxi, nums[i])
        if i >= 2 {
            result = max64(result, int64(nums[i] * diff))
        }
        if i >= 1 {
            diff = max(diff, maxi - nums[i])
        }
    }
    return result
}

func max64(a, b int64) int64 {
    if a > b {
        return a
    }
    return b
}